Sources/Main.cpp

```diff
@@ -370,6 +465,52 @@ const UnicodeString FlagNames[12] = {L"ПАЛУ3", L"!СДЛ1", L"!СДП1", L"!
 const UnicodeString AltFlagNames[12] = {L"П", L"!СДЛ1", L"!СДП1", L"!СДЛ2", L"!СДП2",
     L"РРР0", L"РРР3", L"А15", L"В15", L"П0", L"П1", L"П2"};
 
+bool TX584Form::ParseInput(UnicodeString str, unsigned &Number)
+{
+    int pos = 1;
+    UnicodeString token = NextWord(str, pos);
+    if (token != L"INPUT" && token != L"ВВОД") {
+        return false;
+    }
+
+    UnicodeString NumberString = str.SubString(pos, str.Length()-pos+1);
+    UnicodeString WithoutSpaces = AnsiLowerCase(ReplaceStr(NumberString.Trim(), L" ", L""));
+
+    Number = 0;
+    if (WithoutSpaces.Length() >= 16) {
+        // скорее всего, двоичные коды
+        int count = 0;
+        for (int i = 1; i <= NumberString.Length() && count < 16; i++) {
+            if (NumberString[i] == L' ') {
+                pos++;
+                continue;
+            }
+            if (NumberString[i] != L'0' && NumberString[i] != L'1')
+                return false;
+            Number = (Number << 1) | (NumberString[i] - L'0');
+            count++;
+            pos++;
+        }
+    }
+    else {
+        NumberString = LowerCase(NextWord(str, pos));
+        int SignedNumber;
+        if (!TryStrToInt(NumberString, SignedNumber))
+                return false;
+        if (SignedNumber < -32768 || SignedNumber > 65535) {
+            Number = 65535;
```

Comment:
I would `return false` here instead of hardcoding the number to 65535. This will help to validate user input and display an error message if the number is out of acceptable range.